core-bundle/src/Resources/contao/dca/tl_page.php

```diff
@@ -1119,6 +1120,23 @@ public function scheduleUpdate(Contao\DataContainer $dc)
 		$objSession->set('sitemap_updater', array_unique($session));
 	}
 
+	/**
+	 * Remove already indexed page, if Do not search is checked
+	 */
+	public function removeIndexEntry(Contao\DataContainer $dc)
+	{
+		// Return if there is no ID
+		if (!$dc->activeRecord || !$dc->activeRecord->id) {
+			return;
+		}
+
+		if ($dc->activeRecord->noSearch || $dc->activeRecord->type != 'regular') {
+			$objPage = Contao\PageModel::findById($dc->activeRecord->id);
+			$pageUrl = $objPage->getAbsoluteUrl();
+			Contao\Search::removeEntry($pageUrl);
```

Comment:
I think we should use the delete method of the `contao.search.delegating_indexer` service instead here.

Reply:
No, the `contao.search.indexer`. :)

Reply:
The pages need to be deleted according to the page ID, which is currently not possible via the `contao.search.indexer`.

Reply:
That's incorrect though. For the search indexer we are only allowed to work with external URLs, not page IDs. It's 100% abstracted and must not know anything about any additional meta data.

Reply:
But then it would be impossible to accurately and quickly delete entries from the search index. If you have a page with a newsreader on it, then all the news URLs will need to be purged as well if you enable `noSearch` for the page. How would you gather the URLs to delete then? And the page's original URL will not even be present if `requireItem` is active.

Reply:
Yes, it's impossible currently. And that's correct.

Reply:
Can’t we just update the delete method of the indexer? Because the page ID is known from the JSON-LD so removing all pages that are related to a page ID should be no problem: https://github.com/contao/contao/blob/927a912b96d98499be6b589c32623434d7a30b17/core-bundle/src/Search/Indexer/DefaultIndexer.php#L110-L116

Reply:
Yes, that's what I'll do :)